### PR TITLE
Optimized THQ tagger compatible with STXS workflow

### PR DESCRIPTION
**Pull request by prafullasaha**
This PR mainly contains
1. Pre-selection for tHq category has been loosened.
2. Removed DiPhoton MVA cut from the THQLeptonic tagger.
3. Two MVA training is included in tagger, one to reject ttH background and other to reject all non Higgs background. 
4. Trainings are done separately for 16, 17, 18 sample and path of xml weight files are provided in metaconditions of each year.
5. Threshold values of the training are provided in metaconditions with some preliminary optimization, later we may change bit for separate year. 
6. made it compatible with STXS workflow

### CODE REVIEW

File: Taggers/python/flashggDifferentialPhoIdInputsCorrection_cfi.py [hunk 1]
```diff
@@ -54,7 +54,10 @@ def setup_flashggDifferentialPhoIdInputsCorrection( process, metaConditions ):
     setattr(flashggDifferentialPhoIdInputsCorrection, "photonIdMVAweightfile_EE", cms.FileInPath(str(metaConditions['flashggPhotons']['photonIdMVAweightfile_EE'])))
     setattr(flashggDifferentialPhoIdInputsCorrection, "effAreasConfigFile", cms.FileInPath(str(metaConditions['flashggPhotons']['effAreasConfigFile'])))
     setattr(flashggDifferentialPhoIdInputsCorrection, "is2017", cms.bool(metaConditions['flashggPhotons']['is2017']))
-
+    flashggTHQLeptonicTag.thqCatweightfile_ForNonPeakingBkg = cms.FileInPath(str(metaConditions['THQLeptonicTag']['MVAweights_VsAllBkg']))
```
Comment (simonepigazzini) [inline Taggers/python/flashggDifferentialPhoIdInputsCorrection_cfi.py]: This should not be implemented here but inside flashggTagSequence_cff

**Review by simonepigazzini — COMMENTED** [review]
Check inline comments and make sure that the setup script was run. It looks like some c++ part hae a messed up indentation which is probably due to this command [1] not being run during the repo setup phase.

[1] https://github.com/cms-analysis/flashgg/blob/dev_legacy_runII/setup_flashgg.sh#L65 
this line can be rerun manually to make sure that the indentation hook is setup correctly

File: Taggers/python/flashggDifferentialPhoIdInputsCorrection_cfi.py (hunk 1 above)
Comment (prafullasaha) [reply Taggers/python/flashggDifferentialPhoIdInputsCorrection_cfi.py]: It is done. Now I implemented inside flashggTagSequence_cfi

Comment (prafullasaha) [reply Taggers/python/flashggDifferentialPhoIdInputsCorrection_cfi.py]: For indentation I rerun https://github.com/cms-analysis/flashgg/blob/dev_legacy_runII/setup_flashgg.sh#L65 manually. But indentation are still not proper.

File: Taggers/python/flashggDifferentialPhoIdInputsCorrection_cfi.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 from RecoEgamma.EgammaTools.regressionModifier_cfi import regressionModifier 
 from flashgg.Taggers.globalVariables_cff import globalVariables
-
+from flashgg.Taggers.flashggTags_cff import *
```
Comment (edjtscott) [inline Taggers/python/flashggDifferentialPhoIdInputsCorrection_cfi.py]: Seems unlikely that this is needed - please remove unless it really is

Comment (prafullasaha) [reply Taggers/python/flashggDifferentialPhoIdInputsCorrection_cfi.py]: Thanks Ed. I removed it now. . Initially I implemented training xml paths here, so I added that line. Now I defined training xml paths in flashggTagSequence_cfi.

**Comment by edjtscott** [comment]
hi @prafullasaha and @simonepigazzini - the developments from this PR are now included in [PR 1204](https://github.com/cms-analysis/flashgg/pull/1204) so I will close this one.

**Pull request closed without merging**